app/src/main/java/com/geeksville/mesh/ui/SettingsFragment.kt

```diff
@@ -481,16 +504,24 @@ class SettingsFragment : ScreenFragment("Settings"), Logging {
             }
         }
 
+        addManualDeviceButton()
+
         // get rid of the warning text once at least one device is paired.
         // If we are running on an emulator, always leave this message showing so we can test the worst case layout
         val curRadio = scanModel.selectedAddress
 
-        if (curRadio != null && !MockInterface.addressValid(requireContext(), usbRepository, "")) {
+        if (curRadio != null && !MockInterface.addressValid(
+                requireContext(),
+                usbRepository,
+                ""
+            )
+        ) {
```

Comment:
also please do not reformat lines that you didn't have to change the code on

Reply:
I think Android Studio decided to reformat that for me, and I didn't notice. How helpful. I'll get it fixed.